syntax = "proto3";

package gRPC_microservice;

// The definition of the server for returning data information
service ElectrictyData {
  // Return the information
  rpc GetData (RequestInfo) returns (Data) {}
}

// empty message for the GetData parameter to allow future capability to be appended
message RequestInfo {}

// the response message containing the data
message Data {
  // the titles of the fields
  string a_name = 1;
  string b_name = 2;
  // a list of data pairs
  repeated DataPair data_list = 3;
}

message DataPair {
  // a pair consists of two values
  string a = 1;
  string b = 2;
}
